lib/spack/spack/spec.py

```diff
@@ -1550,6 +1550,16 @@ def dag_hash_bit_prefix(self, bits):
         """Get the first <bits> bits of the DAG hash as an integer type."""
         return base32_prefix_bits(self.dag_hash(), bits)
 
+    @property
+    def name_version_build_hash(self):
+        """A convenience function to print the name, version, and first 7
+        characters for a spec, intended for quick printing to the screen
+        to identify the spec for diff and other purposes. This mimics
+        cmd.display_specs but is a much simpler version to generate and return
+        a string instead of writing to an output.
+        """
+        return "%s@%s/%s" % (self.name, self.version, self.build_hash()[0:7])
```

Comment:
We don't need this function -- caller should just use `spec.format("{name}{@version}{/hash:7}")`.  [See here](https://spack.readthedocs.io/en/latest/spack.html#spack.spec.Spec.format).  For this command, you can also do:

```
spec.format("{name}{@version}{/hash:7}", color=llnl.util.tty.color.get_color_when())
```

to colorize automatically (maybe `spec.format()` should use that instead of `False` for its default to further simplify -- that is likely a separate PR to fix all the places)